[feat] 관심 상품 crud api

seyoung5744 opened the pull request:
### 변경사항
- #7  

AS-IS
-----
**관심 상품 CRUD API**

- 로그인한 사용자만 이용할 수 있습니다.
- 로그인 계정, 상품 ID를 파라미터로 관심 상품 CRUD가 가능합니다.


TO-BE
-----
테스트 코드 완성

seyoung5744 committed:
[feat] 관심 상품 저장 API

상품 아이디를 parameter로 전달하면 DB에서 상품 정보 조회 후 Jpa 연관관계에 의해 DB에 FK로 저장

seyoung5744 committed:
[refactor] 관심 상품 등록 시 Cart 테이블에서 조회되도록 수정

seyoung5744 committed:
[feat] 관심 상품 Delete API

로그인 email과 상품 Id를 기반으로 관심 상품 리스트에서 데이터를 삭제합니다.

seyoung5744 committed:
[feat] 관심 상품 조회 API

로그인 email을 파라미터로 관심 상품 전체 리스트를 조회합니다.

seyoung5744 committed:
[docs] 사용하지 않는 gradle dependencies 수정

mingming-mentor commented on src/main/java/com/zerobase/musinsamonitor/service/CartService.java:
```diff
@@ -0,0 +1,76 @@
+            .build());
+    }
+
+    public void deleteProductFromCart(String deleteNo, String email) {
```
@Transactional을 안걸어줘도 될까요? 10개의 deleteProductId가 있고 이중 3번째꺼가 문제가 있다면 앞에 1번째 2번째 데이터는 삭제가 되고 뒤에 3~10번째는 삭제가 안될것 같아서요. 차라리 다 삭제가 안되거나 가능한것만 삭제가 되는 형태가 좋지않나해서 질문드립니다

seyoung5744 replied:
repository에서 deleteByProductAndEmail 에 대해서만 Transactional을 걸어줘서 각각의 삭제만 신경썼는데
전체적으로 걸어주는 게 맞는 거 같습니다. 지적 감사합니다.

mingming-mentor commented on src/main/java/com/zerobase/musinsamonitor/controller/CartController.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    @PostMapping("/cart/delete")
+    public void deleteProductFromCart(@RequestParam("delete_no") String deleteNo,
```
delete_no을 list 형태로 받으면 어떨까 싶네요. 그럼 추후 deleteNo를 따로 parsing하지 않아도 될것 같아요. 그리고 postMaping사용시에는 request body로 데이터를 넘겨주시는것이 좀더 보편적인 방법입니다

seyoung5744 replied:
넵~보편적인 방법으로 수정하겠습니다.

mingming-mentor reviewed (approved):
수고 많으셨습니다 :) 리뷰드렸어요

seyoung5744 commented:
@mingming-mentor 
질문 사항이 있습니다.

CartService에서 repository로부터 Page<Cart> 와 같이 Pageable을 받아서 사용하는데 이를 테스트할 때는 어떻게 할 수 있나요??
List<Cart>를 만들고 Page로 변환을 해야하나요??

seyoung5744 committed:
[rename] 응답 dto 패키지명 responsedto로 수정 및 model 패키지 하위로 이동

seyoung5744 committed:
[refactor] 관심 상품 삭제 API 수정

1. RequestBody로 상품 id 리스트를 받도록 수정
2. 각 delete 쿼리에만 있던 Transaction을 CartService에서 select, delete  Transaction으로 수정
